prusa/link/web/main.py

```diff
@@ -476,6 +476,21 @@ def api_update(req, env):
     return Response(status_code=state.HTTP_BAD_REQUEST)
 
 
+@app.route("/api/v1/update/<env>", method=state.METHOD_POST)
+@check_api_digest
+def api_update_post(req, env):
+    """Update all packages in given environment"""
+    # pylint: disable=unused-argument
+    if env == "python":
+        call("pip install --upgrade " + ' '.join(working_set), shell=True)
```

Comment:
I need more explanation, what this exactly do

Reply:
To me this looks like it has pip attempt to update all the python packages sitting in the working_set.

From these [docs](https://setuptools.pypa.io/en/latest/pkg_resources.html)

`working_set`
A collection of distributions actually available for importing, as on sys.path. At most one distribution (release version) of a given project may be present in a working set, as otherwise there would be ambiguity as to what to import.

`Use of pkg_resources is discouraged in favor of [importlib.resources](https://docs.python.org/3/library/importlib.html#module-importlib.resources), [importlib.metadata](https://docs.python.org/3/library/importlib.metadata.html), and their backports ([importlib_resources](https://pypi.org/project/importlib_resources), [importlib_metadata](https://pypi.org/project/importlib_metadata)). Please consider using those libraries instead of pkg_resources.
`

Based on the above suggestion `importlib.metadata` good replacement.